drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-backend/src/test/java/org/drools/workbench/screens/testscenario/backend/server/ScenarioTestEditorServiceImplTest.java

```diff
@@ -314,6 +314,37 @@ public void testCopyWithTargetThrowException() throws Exception {
     }
 
     @Test
+    public void testRunScenarioDependentImportsAndFactData() throws Exception {
+        final ArrayList<Fixture> fixtures = new ArrayList<Fixture>();
+        final Imports imports = new Imports() {{
+            addImport(new Import("java.sql.ClientInfoStatus"));
+            addImport(new Import("Integer"));
+            addImport(new Import("org.kie.Person"));
+        }};
+
+        final Map<String, ModelField[]> modelFields = new HashMap<String, ModelField[]>() {{
+            put("java.sql.ClientInfoStatus",
+                new ModelField[]{modelField("java.sql.JDBCType")});
+            put("org.kie.Person",
+                new ModelField[]{modelField("org.kie.Address")});
+            put("org.kie.Address",
+                new ModelField[]{modelField("org.kie.City"), modelField("org.kie.Street")});
+        }};
+
+        when(scenario.getFixtures()).thenReturn(fixtures);
+        when(dataModelService.getDataModel(path)).thenReturn(modelOracle);
+        when(modelOracle.getProjectModelFields()).thenReturn(modelFields);
+        when(scenario.getImports()).thenReturn(imports);
+
+        testEditorService.runScenario("userName",
+                                      path,
+                                      scenario);
+
+        assertEquals(7,
```

Comment:
`Assertions.assertThat(scenario.getImports().getImports()).hasSize(7);`

Also, wouldn't be bad to name the constant, so that the future generations know why 7.